pkg/webhooks/pod/pod.go

```diff
@@ -0,0 +1,174 @@
+	log                   = logf.Log.WithName(WebhookName)
+
+	scope = admissionregv1.NamespacedScope
+	rules = []admissionregv1.RuleWithOperations{
+		{
+			Operations: []admissionregv1.OperationType{admissionregv1.OperationAll},
+			Rule: admissionregv1.Rule{
+				APIGroups:   []string{"v1"},
+				APIVersions: []string{"*"},
+				Resources:   []string{"pods"},
+				Scope:       &scope,
+			},
+		},
+	}
+)
+
+type PodWebhook struct {
+	mu sync.Mutex
+	s  runtime.Scheme
+}
+
+// TimeoutSeconds implements Webhook interface
+func (s *PodWebhook) TimeoutSeconds() int32 { return 1 }
+
+// MatchPolicy implements Webhook interface
+func (s *PodWebhook) MatchPolicy() admissionregv1.MatchPolicyType {
+	return admissionregv1.Equivalent
+}
+
+// Name implements Webhook interface
+func (s *PodWebhook) Name() string { return WebhookName }
+
+//FailurePolicy defines how unrecognized errors and timeout errors from the admission webhook are handled. Allowed values are Ignore or Fail.
+//Ignore means that an error calling the webhook is ignored and the API request is allowed to continue.
+//It's important to leave the FailurePolicy set to Ignore because otherwise the pod will fail to be created as the API request will be rejected.
+func (s *PodWebhook) FailurePolicy() admissionregv1.FailurePolicyType {
+	return admissionregv1.Ignore
+}
+
+// Rules implements Webhook interface
+func (s *PodWebhook) Rules() []admissionregv1.RuleWithOperations { return rules }
+
+// GetURI implements Webhook interface
+func (s *PodWebhook) GetURI() string { return "/" + WebhookName }
+
+// SideEffects implements Webhook interface
+func (s *PodWebhook) SideEffects() admissionregv1.SideEffectClass {
+	return admissionregv1.SideEffectClassNone
+}
+
+// Validate implements Webhook interface
+func (s *PodWebhook) Validate(req admissionctl.Request) bool {
+	valid := true
+	valid = valid && (req.UserInfo.Username != "")
+	valid = valid && (req.Kind.Kind == "Pod")
+
+	return valid
+}
+
+func (s *PodWebhook) renderPod(req admissionctl.Request) (*corev1.Pod, error) {
+	decoder, err := admissionctl.NewDecoder(&s.s)
+	if err != nil {
+		return nil, err
+	}
+	pod := &corev1.Pod{}
+	if len(req.OldObject.Raw) > 0 {
+		err = decoder.DecodeRaw(req.OldObject, pod)
+	} else {
+		err = decoder.DecodeRaw(req.Object, pod)
+	}
+	if err != nil {
+		return nil, err
+	}
+	return pod, nil
+}
+
+func (s *PodWebhook) authorized(request admissionctl.Request) admissionctl.Response {
+	var ret admissionctl.Response
+	pod, err := s.renderPod(request)
+	if err != nil {
+		log.Error(err, "Couldn't render a Pod from the incoming request")
+		return admissionctl.Errored(http.StatusBadRequest, err)
+	}
+	//If the incoming Pod is aimed at a privileged namespace except for openshift-logging and openshift-operators, allow it to do whatever it wants.
+	//However, if the pod is targeting a customer's namespace (aka non-privileged), then it may not tolerate certain master/infra node taints.
```

Comment:
```suggestion
	// If the incoming Pod is aimed at a privileged namespace except for exceptionNamespace, allow it to do whatever it wants.
	// However, if the pod is targeting a customer's namespace (aka non-privileged), then it may not tolerate certain master/infra node taints.
```